Add support for classical return values in one-shot mode (#2004)
PennyLaneAI/catalyst: doc/releases/changelog-dev.md

<h3>Improvements 🛠</h3>

> * Catalyst now supports classical return values with one-shot mode.

Reviewer: ```suggestion
* Catalyst now supports returning classical and MCM values with the dynamic one-shot MCM method.
```

>   For example, the code below will generate 10 values, with an equal probability of 42 and 43
>   appearing. You can also return `measure(0)` as well.
>   [(#2001)](https://github.com/PennyLaneAI/catalyst/pull/2001)
> 
>   ```python
>   from catalyst import qjit, measure
>   @qjit(autograph=True)
>   @qml.set_shots(10)
>   @qml.qnode(qml.device("lightning.qubit", wires=1), mcm_method="one-shot")
>   def circuit():
>       qml.Hadamard(wires=0)
>       if measure(0):
>           return 42
>       else:
>           return 43
> 
>   print(circuit())
>   ```
> 
>   which produces either 42 or 43 with equal probability (approximately 50% each)
> 
>   ```pycon
>   [43 42 42 42 43 42 43 42 43 42]
>   ```

Reviewer: ````suggestion
  [(#2001)](https://github.com/PennyLaneAI/catalyst/pull/2001)
  
  For example, the code below will generate 10 values, with an equal probability of 42 and 43
  appearing. 

  ```python
  import pennylane as qml
  from catalyst import qjit, measure
  
  @qjit(autograph=True)
  @qml.set_shots(10)
  @qml.qnode(qml.device("lightning.qubit", wires=1), mcm_method="one-shot")
  def circuit():
      qml.Hadamard(wires=0)
      m = measure(0)
      if m:
          return 42, m
      else:
          return 43, m
  ```

  ```pycon
  >>>  print(circuit())
  (Array([42, 43, 42, 42, 43, 42, 42, 43, 42, 42], dtype=int64),
   Array([ True, False,  True,  True, False,  True,  True, False,  True,
           True], dtype=bool))

  ```
````

Reviewer: Shows off both features (classical vals and MCM vals in return statement)!

<h3>Improvements 🛠</h3>

* Catalyst now supports returning classical and MCM values with the dynamic one-shot MCM method.
  [(#2001)](https://github.com/PennyLaneAI/catalyst/pull/2001)
  
  For example, the code below will generate 10 values, with an equal probability of 42 and 43
  appearing. 

  ```python
  import pennylane as qml
  from catalyst import qjit, measure
  
  @qjit(autograph=True)
  @qml.set_shots(10)
  @qml.qnode(qml.device("lightning.qubit", wires=1), mcm_method="one-shot")
  def circuit():
      qml.Hadamard(wires=0)
      m = measure(0)
      if m:
          return 42, m
      else:
          return 43, m
  ```

  ```pycon
  >>>  print(circuit())
  (Array([42, 43, 42, 42, 43, 42, 42, 43, 42, 42], dtype=int64),
   Array([ True, False,  True,  True, False,  True,  True, False,  True,
           True], dtype=bool))

> 

Reviewer: ```suggestion
```
